### DISCUSSION TITLE
Is there a way to always have `request` and `response` available in all handlers?

### DISCUSSION THREAD

**Discussion by TommasoAmici** [fastapi/fastapi#5821] [Questions] [answered]
Hi all,
I'm implementing a decorator to cache responses in Redis. It all works well, but since I need access to the `request: Request` and `response: Response` objects I now have to manually include it in all handlers even though they won't be used in those functions at all.

```py
@app.post("/xxx")
@cache_view()
def xxx(request: Request, response: Response):
    return {1: 2}
```
Is there a way I can _always_ have request and response in `*args` or `**kwargs`?

**Comment by jgould22** [comment, marked as answer]
It is for that reason that people usually implement this type of thing with an ASGI middle-ware. 

You would have to configure the middle-ware though as it will apply to all routes.

A more FastAPI like option would probably be to use a route based dependency that could be added in the route decorator.

**Reply by TommasoAmici** [reply]
Thanks for your answer, I'll check if what I have can be re-implemented as a middleware instead
